### PR TITLE
Migrate OPA from glide to go mod

### PR DESCRIPTION
**Pull request by patrick-east**
This PR migrates OPA to be a Go module instead of using glide. The changes are batched together and include:

* Initializing and module and swapping from glide to mod vendored packages.
* Switching to use a tools.go w/ imports to pin tool versions (goimports, pigeon, etc).
* Updating make targets to forcibly turn on module mode and use the vendor directory.
* Updating make targets to use `go run ...` instead of `install-deps.sh` and binaries directly.
* Updating dev docs to explain how to update dependencies.

Some high level benefits of this:

* Updating individual packages is super easy, a PR that does just that for goimports is ready as soon as this merges.
* Easier for newer module based projects to use OPA as a library. This is the future direction for most golang projects. Glide is dead.. long live modules.

More details: https://github.com/golang/go/wiki/Modules (the FAQ section covers a _ton_ of stuff).

Fixes: #1556

### CODE REVIEW

**Comment by patrick-east** [comment]
One note for compatibility testing: I created a test project that used glide and pulled the opa module in as a dependency, everything worked as expected. I also updated a dep project (https://github.com/open-policy-agent/frameworks/tree/master/constraint) and it was able to pull in opa w/ its vendored stuff pretty happily as well.

File: tools.go [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// +build tools
+
+// Copyright 2019 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package tools
+
+// See build/install-deps.sh for a list of required tools
```
Comment (ashutosh-narkar) [inline tools.go]: NIT: `build/install-deps.sh` is deleted.

Comment (patrick-east) [reply tools.go]: Fixed in latest updates.

File: Makefile [hunk 2]
```diff
@@ -4,7 +4,9 @@
 
 VERSION := 0.15.0-dev
 
-GO := go
+# Always use the vendor directory. Remove this
+# when the OPA vendor directory is removed.
+GO := GOFLAGS=-mod=vendor go
```
[the remark this answers is not in the document]
Comment (patrick-east) [reply Makefile]: I didn't with the rational that the instructions are for making some new project and importing OPA. If someone was to fork OPA and customize in-place they would probably want to build that way but the instructions are (I think) guiding down a different path where OPA is a dependency (right?)

If someone does fork OPA they should, IMO, just use the makefiles as normal.

**Pull request merged**